src/controllers/rings-by-controller.ts

```diff
@@ -21,10 +30,22 @@ export default class RingsByController extends AbstractController {
 
   @GET
   @Path('/')
-  @Response<Ring[]>(200, 'List of all available rings.')
+  @Response<RingListResponse>(200, 'List of all available rings.')
   @Response<CustomError>(401, 'Unauthorised.')
-  public async findRings(): Promise<Ring[]> {
-    return this.rings.find();
+  public async findRings(
+    @QueryParam('pageNumber') pageNumber?: number,
+    @QueryParam('pageSize') pageSize?: number,
+    @QueryParam('sortingDirection') sortingDirection?: SortingDirection,
+    @QueryParam('sortingColumn') sortingColumn?: string,
+  ): Promise<RingListResponse> {
+    const paramsSearch = parsePageParams({ pageNumber, pageSize, sortingColumn, sortingDirection });
+    const [rings, totalElements] = await this.rings.findAndCount(paramsSearch);
+    return {
+      content: rings,
+      pageNumber: paramsSearch.number,
+      pageSize: paramsSearch.size,
+      totalElements,
+    };
```

Comment:
it's a little weird to put `longVariable` in function to then get `shortVariable`  back to then remap it in `longVariable`